File: Backend/CaffShop.Shared/DTOs/UserDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CaffShop.Shared.DTOs
{
  public class UserDto
  {
    public int Id { get; set; }
    public string Name { get; set; }
    public string UserName { get; set; }
    public List<CaffListItemDto> Caffs { get; set; }
  }
}

File: Frontend/lib/models/user_dto.dart
class UserDto {
  final int id;
  String? name;
  String? username;
  List? caffs;

  UserDto({required this.id, this.name, this.username, this.caffs});

  factory UserDto.fromJson(Map<String, dynamic> json) => UserDto(
      id: json['id'],
      name: json['name'],
      username: json['userName'],
      caffs: json['caffs']);

  Map<String, dynamic> toJson() =>
      {'id': id, 'name': name, 'username': username, 'caffs': caffs};
}
